spot

```diff
@@ -1,64 +1,59 @@
 #!/bin/bash
-if [[ $# != 1 ]]
-then
-	printf "Usage\n\tspot <URL>\n"
-	printf "\nExample\n\thttp://open.spotify.com/user/1248318865/playlist/7uKPSEwUQzx3c9EU032lU6\n"
-	exit 0
-fi
 
-input=$1
-keyword=spotify
-schemedurl=
-url=
+# enable shell script debugging
+# set -x
 
-matches() {
-    input="$1"
-    pattern="$2"
-    echo "$input" | grep -q "$pattern"
-}
-
-if matches $input $keyword
+# display help if incorrect usage or on --help
+if [[ ($# == 0) || ($1 == "--help") ]]
 then
-        url=$input
-else
-        echo "$1 is not a valid spotify url" 
-	exit 0;
+    printf "Usage\n\tspot <spotify web url OR search string>\n\tspot --help to see this help text\n"
+    printf "\nExamples\n\tspot http://open.spotify.com/user/1248318865/playlist/7uKPSEwUQzx3c9EU032lU6\n"
+    printf "\tspot breaking benjamin\n"
+    printf "\tspot --help\n"
+    exit 0
 fi
 
-IFS='\/ ' read -r -a array <<< "$url";
-i=0;
-foundCom=0;
-arrLen=${#array[@]}
-word=com
-while [ $i != $arrLen ]
-do
-        if [ "${array[$i]#*$word}" != "${array[$i]}" ]
-        then
-                foundCom=1;
-                i=`expr $i + 1`;
-        fi
-        if [ $foundCom -eq 0 ]
-        then
-                #echo "display dialog \"Not printing ${array[i]}\"";
-                i=`expr $i + 1`;
-                continue;
-        else
-                schemedurl="$schemedurl:${array[$i]}"
-                #echo "display dialog \"Printing ${array[i]}\"";
-                i=`expr $i + 1`;
-        fi
-done
-FINSTR=spotify$schemedurl
-
-osascript<<EOF
+# meat of the script
+# use apple script to put a string in spotify app search box
+_runSpotOsaScript() {
+    osascript<<EOF
 tell application "spotify" to activate
+delay 0.5
```

Comment:
This delay compensates for switch between applications.
It's very subjective but 0.5 seems to hit a sweet spot. Might cause issues for other users.